### PR TITLE
PMIP fix issue with mono_dont_free_domains

### PR DESCRIPTION
**Pull request by mderoy**

### CODE REVIEW

File: PmipCallStackFilter.cs [hunk 1]
```diff
@@ -50,70 +56,78 @@ public static DkmStackWalkFrame PmipStackFrame(DkmStackContext stackContext, Dkm
...
-            {
-                if (!File.Exists(fileName))
-                    return;
+            DirectoryInfo taskDirectory = new DirectoryInfo(Path.GetTempPath());
+            FileInfo[] taskFiles = taskDirectory.GetFiles("pmip_" + pid + "_*.txt");
+
+            if (taskFiles.Length < 1)
+                return;
+
+            Array.Sort(taskFiles, (a, b) => Path.GetFileNameWithoutExtension(a.Name).CompareTo(Path.GetFileNameWithoutExtension(b.Name)));
```
Comment (joncham) [inline PmipCallStackFilter.cs]: Will this correctly compare Foo1.txt and Foo10.txt?

File: PmipCallStackFilter.cs [hunk 2]
```diff
@@ -50,70 +56,78 @@ public static DkmStackWalkFrame PmipStackFrame(DkmStackContext stackContext, Dkm
...
-                            var endip = tokens[1];
-                            var description = tokens[2];
-
-                            var startiplong = ulong.Parse(startip, NumberStyles.HexNumber);
-                            var endipint = ulong.Parse(endip, NumberStyles.HexNumber);
-
-                            list.Add(new Range() { Name = description, Start = startiplong, End = endipint });
-                        }
-                    }
+                    Console.WriteLine("Unable to read dumped pmip file: " + ex.Message);
```
Comment (joncham) [inline PmipCallStackFilter.cs]: Does this go somewhere useful?

File: PmipCallStackFilter.cs [hunk 3]
```diff
@@ -50,70 +56,78 @@ public static DkmStackWalkFrame PmipStackFrame(DkmStackContext stackContext, Dkm
...
-
         }
 
         public static bool TryGetDescriptionForIp(ulong ip, out string name)
         {
             name = string.Empty;
 
-            if (_rangesSortedByIp == null)
-                return false;
+            if(_count != _rangesSortedByIp.Count)
```
Comment (joncham) [inline PmipCallStackFilter.cs]: Do we want to sort here now, or put it back with the file reading? Then we can drop the `_count` variable I think.

File: PmipCallStackFilter.cs (hunk 1 above)
Comment (mderoy) [reply PmipCallStackFilter.cs]: Probably not. I'll instead parse the number out of the filename and compare those as integers.

File: PmipCallStackFilter.cs (hunk 2 above)
Comment (mderoy) [reply PmipCallStackFilter.cs]: No. I think theres a log function to print to the output window in VS...I'll take a look

File: PmipCallStackFilter.cs (hunk 3 above)
Comment (mderoy) [reply PmipCallStackFilter.cs]: Sorting after reading the file is nicer I'd say :)

**Comment by mderoy** [comment]
@joncham made some fixes based on your suggestions

**Review by joncham — APPROVED** [review]

**Pull request merged**